optpresso/server/static/js/app.js

```diff
@@ -0,0 +1,394 @@
+    for (let [key, value] of Object.entries(this.state.config)) {
+      if (value == null) {
+        value = ""
+      }
+      form.append(key, value);
+    }
+    httpPostAsync("/config/", form, function(){
+      event.target.disabled = false;
+    })
+  }
+
+  handleChange(event) {
+    let val = event.target.value
+    if (event.target.type == "checkbox") {
+      val = event.target.checked
+    }
+    if (this.state.config.hasOwnProperty(event.target.name)) {
+      this.setState({
+        config: {
+          ...this.state.config,
+          [event.target.name]: val,
+        },
+      })
+    } else {
+      this.setState({
+        pullData: {
+          ...this.state.pullData,
+          [event.target.name]: val,
+        },
+      })
+    }
+  }
+
+  handleSubmit(event) {
+    event.target.disabled = true;
+    this.props.setConfig(this.state.config);
+    event.preventDefault();
+    captureImage(
+      {...this.state.config, ...this.state.pullData},
+      function(data) {event.target.disabled = false},
+    )
+  }
+
+  render() {
+    let configInputs = [];
+    for (let key in this.state.config) {
+      let value = this.state.config[key]
+      let labelName = key.replace(' ', '%nbsp;')
+      configInputs.push(
+        <div class="form-group">
+          <label>
+            {labelName}:
+            <input type="text" name={key} value={value} onChange={evt => this.handleChange(evt)} />
+          </label>
+        </div>
+      );
+    }
+    let captureInputs = [];
+    for (let key in this.state.pullData) {
+      let value = this.state.pullData[key]
+      let labelName = key.replace(' ', '%nbsp;')
+      if (typeof value === "boolean") {
+        captureInputs.push(
+          <div class="checkbox">
+            <label>
+              {labelName}:
+              <input name={key} type="checkbox" checked={value} onClick={evt => this.handleChange(evt)}/>
+            </label>
+          </div>
+        )
+      } else {
+        captureInputs.push(
+          <div class="form-group">
+            <label>
+              {labelName}:
+              <input type="text" name={key} value={value} onChange={evt => this.handleChange(evt)} />
+            </label>
+          </div>
+        )
+      }
+    }
+    return (
+      <div class="container">
+        <div class="row">
+          <div class="col-md-5 offset-1 config">
+            <h3>Configuration</h3>
+            {configInputs}
+            <button onClick={evt => this.handleConfigUpdate(evt)}>Update Config</button>
+          </div>
+          <div class="col-md-5 offset-1 config">
+            <h3>Capture</h3>
+            {captureInputs}
+            <button onClick={evt => this.handleSubmit(evt)}>Capture</button>
+          </div>
+        </div>
+      </div>
+    )
+  }
+}
+
+function CaptureComponent() {
+  const [mode, setMode] = useGlobal("mode");
+  const [config, setConfig] = useGlobal("config");
+  const getExistingConfig = async () => {
```

Comment:
this func doesn't need to be async since its internal methods aren't `await`ing anything